standard_revisions/match_variables_from_two_versions_of_a_dataset.py

```diff
@@ -0,0 +1,313 @@
+                mapping = pd.concat([mapping, mapping_added], ignore_index=True)
+
+    return mapping
+
+
+def display_summary(old_variables, new_variables, mapping):
+    """Display summary of the result of the mapping.
+
+    Parameters
+    ----------
+    old_variables : pd.DataFrame
+        Table of old variable names (column 'name') and ids (column 'id').
+    new_variables : pd.DataFrame
+        Table of new variable names (column 'name') and ids (column 'id').
+    mapping : pd.DataFrame
+        Mapping table from old variable name and id to new variable name and id.
+
+    """
+    print("Matched pairs:")
+    for i, row in mapping.iterrows():
+        print(f"\n  {row['name_old']} ({row['id_old']})")
+        print(f"  {row['name_new']} ({row['id_new']})")
+
+    unmatched_old = old_variables[~old_variables['name'].isin(mapping['name_old'])].reset_index(drop=True)
+    unmatched_new = new_variables[~new_variables['name'].isin(mapping['name_new'])].reset_index(drop=True)
+    if len(unmatched_old) > 0:
+        print("\nUnmatched variables in the old dataset:")
+        for i, row in unmatched_old.iterrows():
+            print(f"  {row['name']} ({row['id']})")
+    else:
+        print("\nAll variables in the old dataset have been matched.")
+    if len(unmatched_new) > 0:
+        print("\nUnmatched variables in the new dataset:")
+        for i, row in unmatched_new.iterrows():
+            print(f"  {row['name']} ({row['id']})")
+    else:
+        print("\nAll variables in the new dataset have been matched.")
+
+
+def save_variable_replacements_file(mapping, output_file=OUTPUT_FILE):
+    """Save a json file with the mapping from old to new variable ids.
+
+    Parameters
+    ----------
+    mapping : pd.DataFrame
+        Mapping table from old variable name and id to new variable name and id.
+    output_file : str
+        Path to output file.
+
+    """
+    # Create a dictionary mapping from old variable id to new variable id.
+    mapping_indexes = mapping[['id_old', 'id_new']].set_index('id_old').to_dict()['id_new']
+    mapping_indexes = {str(key): str(mapping_indexes[key]) for key in mapping_indexes}
+
+    print(f"Saving index mapping to json file: {output_file}")
+    save_data_to_json_file(data=mapping_indexes, json_file=output_file, **{'indent': 4, 'sort_keys': True})
+
+
+def main(old_dataset_name, new_dataset_name, omit_identical=OMIT_IDENTICAL, matching_function=MATCHING_FUNCTION,
+         output_file=OUTPUT_FILE):
+    with get_connection() as db_conn:
+        # Get old and new dataset ids.
+        old_dataset_id = get_dataset_id(db_conn=db_conn, dataset_name=old_dataset_name)
+        new_dataset_id = get_dataset_id(db_conn=db_conn, dataset_name=new_dataset_name)
+
+        # Get variables for old and new datasets.
+        old_variables = get_variables_in_dataset(db_conn=db_conn, dataset_id=old_dataset_id)
```

Comment:
This should be revised to only retrieve old variables that are used in at least 1 chart, since the output `variable_replacements.json` is only consumed by the `ChartRevisionSuggester`. Fuzzy matching old variables -> new variables for old variables that are not used in a chart is not needed. 

e.g. World Bank WDI contains >1400 variables, of which only something like 30-40% are used in charts.